packages/lodestar/src/tasks/index.ts

```diff
@@ -70,8 +74,27 @@ export class TasksService implements IService {
     await this.interopSubnetsTask.stop();
   };
 
-  private handleFinalizedCheckpointChores = async (finalized: BlockSummary, pruned: BlockSummary[]): Promise<void> => {
-    await new ArchiveBlocksTask(this.config, {db: this.db, logger: this.logger}, finalized, pruned).run();
-    await new ArchiveStatesTask(this.config, {db: this.db, logger: this.logger}, finalized, pruned).run();
+  private onFinalizedCheckpoint = async (finalized: Checkpoint): Promise<void> => {
+    await new ArchiveStatesTask(this.config, {db: this.db, logger: this.logger}, finalized).run();
```

Comment:
we want to run ArchiveStateTask after ArchiveBlockTask so that when we restart, we start from last finalized state and we want to make sure finalized blocks stored in db already